rest/oidc_api.go

```diff
@@ -95,10 +95,11 @@ func (h *handler) handleOIDCCommon() (redirectURLString string, err error) {
 		return redirectURLString, err
 	}
 
-	client := provider.GetClient(h.getOIDCCallbackURL)
-	if client == nil {
+	client, err := provider.GetClient(h.getOIDCCallbackURL)
+	if err != nil {
+		base.Errorf("OIDC initialization error: %v", err)
 		return redirectURLString, base.HTTPErrorf(
-			http.StatusInternalServerError, fmt.Sprintf("Unable to obtain client for provider:%s", providerName))
+			http.StatusInternalServerError, fmt.Sprintf("Unable to obtain client for provider: %s", providerName))
```

Comment:
Include error message for the user making the request, now that we have access to it

```suggestion
			http.StatusInternalServerError, fmt.Sprintf("Unable to obtain client for provider: %s - %v", providerName, err))
```

Reply:
Will do. Originally didn't do this because I thought the error message wasn't the most helpful but on second thoughts it could help.
Example error message when OIDC Provider Unreachable: `Unable to obtain client for provider: foo - unable to discover config: unsuccessful response: <nil>`

Reply:
Ah right - yeah that `<nil>` is a bit misleading. Looks like another usage of an incorrect "err" variable...

```
	if resp.StatusCode != http.StatusOK {
		return ProviderMetadata{}, MaxProviderConfigSyncInterval, false, fmt.Errorf("unsuccessful response: %v", err)
	}
```

I think there we should include the repsonse body, in the event that the provider is giving us more detail in the error repsonse.